python/llm/src/bigdl/llm/transformers/low_bit_linear.py

```diff
@@ -308,13 +360,20 @@ def to(self, *args, **kwargs):
             self.data = ggml_q_format_convet_cpu2xpu(self.data,
                                                      reduce(mul, self._shape, 1),
                                                      self.qtype)
+            if self.enable_xetla:
+                self.data = q4_0_xpu_transpose(self.data, self._shape)
             new_param = FP4Params(super().to(device=device,
                                              dtype=dtype,
                                              non_blocking=non_blocking),
                                   requires_grad=self.requires_grad,
                                   quantized=self.quantized,
                                   _shape=self._shape,
-                                  qtype=self.qtype)
+                                  qtype=self.qtype,
+                                  enable_xetla=self.enable_xetla)
+            if self.enable_xetla:
```

Comment:
Can we merge the two `if`s?

Reply:
`get_xpu_device_type` requires the `data` on `xpu`. The `data` is on `xpu` under the first if.